arch/arm/core/aarch64/fatal.c

```diff
@@ -179,11 +194,18 @@ void z_arm64_fatal_error(unsigned int reason, const z_arch_esf_t *esf)
 		}
 
 		if (GET_EL(el) != MODE_EL0) {
-			LOG_ERR("ESR_ELn: 0x%016llx", esr);
-			LOG_ERR("FAR_ELn: 0x%016llx", far);
+			bool recoverable = false;
+			bool dump_far = false;
+
 			LOG_ERR("ELR_ELn: 0x%016llx", elr);
 
-			print_EC_cause(esr);
+			dump_esr(esf, esr, &dump_far, &recoverable);
```

Comment:
it might be slightly better to separate the logic which dumps information, from the logic which determines whether the exception is recoverable.

I am not sure if in the current code here, whether gc-sections will discard all the strings if LOG is turned off. On x86 I had to ifdef around CONFIG_EXCEPTION_DEBUG to make sure it was.

Reply:
> it might be slightly better to separate the logic which dumps information, from the logic which determines whether the exception is recoverable.

Ok

> I am not sure if in the current code here, whether gc-sections will discard all the strings if LOG is turned off. On x86 I had to ifdef around CONFIG_EXCEPTION_DEBUG to make sure it was.

Indeed the strings are not discarded when `CONFIG_LOG=n`. I'll add a `CONFIG_EXCEPTION_DEBUG` for AArch64 as well.